pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -66,7 +68,25 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 		return existingConfig, append(errs, err)
 	}
 
+	featureGate, err := listers.FeatureGateLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		recorder.Eventf("ObserveCloudProviderNames", "Optional featuregate.%s/cluster not found", configv1.GroupName)
+	} else if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	external, err := cloudprovider.IsCloudProviderExternal(infrastructure.Status.Platform, featureGate)
+	if err != nil {
+		recorder.Eventf("ObserveCloudProviderNames", "Invalid featuregate.%s/cluster format: %v", err)
```

Comment:
This error description seems oddly specific given its for a call `IsCloudProviderExternal`, would it not make more sense for it to say something like `Could not determine external cloud provider state: %v`?